hello-world/tutorial/index.md

```diff
@@ -0,0 +1,494 @@
+
+```go
+// planet/x/blog/keeper/ibcPost.go
+// OnRecvIbcPostPacket processes packet reception
+func (k Keeper) OnRecvIbcPostPacket(ctx sdk.Context, packet channeltypes.Packet, data types.IbcPostPacketData) (packetAck types.IbcPostPacketAck, err error) {
+	// validate packet data upon receiving
+	if err := data.ValidateBasic(); err != nil {
+		return packetAck, err
+	}
+
+	packetAck.PostID = k.AppendPost(
+		ctx,
+		packet.SourcePort+"-"+packet.SourceChannel+"-"+data.Creator,
+		data.Title,
+		data.Content,
+	)
+
+	return packetAck, nil
+}
+```
+
+### Receive the Post Acknowledgement
+
+On the sending blockchain, store a `sentPost` so you know that the post has been received on the target chain.
+
+Store the title and the target to identify the post.
+
+When a packet is scaffolded, the default type for the received acknowledgment data is a type that identifies if the packet treatment has failed. The `Acknowledgement_Error` type is set if `OnRecvIbcPostPacket` returns an error from the packet.
+
+```go
+// OnAcknowledgementIbcPostPacket responds to the the success or failure of a packet
+// acknowledgement written on the receiving chain.
+func (k Keeper) OnAcknowledgementIbcPostPacket(ctx sdk.Context, packet channeltypes.Packet, data types.IbcPostPacketData, ack channeltypes.Acknowledgement) error {
+	switch dispatchedAck := ack.Response.(type) {
+	case *channeltypes.Acknowledgement_Error:
+		// We will not treat acknowledgment error in this tutorial
+		return nil
+	case *channeltypes.Acknowledgement_Result:
+		// Decode the packet acknowledgment
+		var packetAck types.IbcPostPacketAck
+		err := packetAck.Unmarshal(dispatchedAck.Result)
+		if err != nil {
+			// The counter-party module doesn't implement the correct acknowledgment format
+			return errors.New("cannot unmarshal acknowledgment")
+		}
+
+		k.AppendSentPost(
+			ctx,
+			data.Creator,
+			packetAck.PostID,
+			data.Title,
+			packet.DestinationPort+"-"+packet.DestinationChannel,
+		)
+
+		return nil
+	default:
+		// The counter-party module doesn't implement the correct acknowledgment format
+		return errors.New("invalid acknowledgment format")
+	}
+}
+```
+
+### Store Information about the Timed-out Packet
+
+Store posts that have not been received by target chains in `timedoutPost` posts. This logic follows the same format as `sentPost`.
+
+```go
+// OnTimeoutIbcPostPacket responds to the case where a packet has not been transmitted because of a timeout
+func (k Keeper) OnTimeoutIbcPostPacket(ctx sdk.Context, packet channeltypes.Packet, data types.IbcPostPacketData) error {
+
+	k.AppendTimedoutPost(
+		ctx,
+		data.Creator,
+		data.Title,
+		packet.DestinationPort+"-"+packet.DestinationChannel,
+	)
+
+	return nil
+}
+```
+
+This is the basic `blog` module setup.
+The blockchain is now ready!
+
+Spin up the blockchain and send a blog post from one blockchain app to the other.
+
+Various terminal windows are required to complete these next steps.
```

Comment:
```suggestion
You will need two terminal windows open for the next steps.
```

Reply:
I think it is two for the blockchains, one more for the relayer and then one to enter the commands?

Reply:
Then "Multiple terminal windows are required to complete these next steps." I just don't think "various" is the right word here.